lib/bazel/affected_targets.go

```diff
@@ -102,13 +105,24 @@ func GetAffectedTargets(start string, end string, config *ppb.PresubmitConfig, l
 			var err error
 			endResults, err = endWorkspace.Query("deps(//...)", WithUnorderedOutput(), workspaceLogEnd)
 			if err != nil {
-				return fmt.Errorf("failed to query deps for end point: %w", err)
+				endQueryErr = fmt.Errorf("failed to query deps for end point: %w", err)
+				return endQueryErr
 			}
 			log.Infof("Queried info for %d targets from 'after' workspace", len(endResults.Targets))
 			return nil
 		},
 	)
 	if errs != nil {
+		if startQueryErr != nil && endQueryErr == nil {
+			// We are calculating targets over a change that fixes the build graph
+			// (broken before, working after). In a presubmit context, we want this
+			// step to succeed, but there is no sensible list of targets that the
+			// change affects since the build graph was broken in one stage.
+			//
+			// Pass here but emit a warning.
+			log.Warnf("Broken build graph detected at start point; this change fixes the build graph, but no targets will be tested. This change must be tested manually.")
```

Comment:
I think this is a good improvement, let's merge it in as is so that we have a fix for this problem.

However, my recommendation here would be:
1) Don't emit a warning, rather return a specific error type - so that the caller can tell that the problem was that the source graph could not be built, but the dest graph could.
2) Have the caller decide if the error is worth dying for, or just emit warning. Personally, I would a) emit warning, BUT b) exit with a specific status code, eg, 120.

The tool invoking the diff can then see that the problem is the graph being restored, and implement a different strategy to test the PR. For example: just look at the files modified, and run the tests corresponding to the modified files. Or ignore the problem for now.

Reason: I'm worried about rollbacks or PRs fixing the graph, but also touching other files. Again, this is a relatively little gain compared to the overall gain of this PR, so feel free to merge as is (or maybe just have the different error type? let the caller decide).